props/hev.h

```diff
@@ -0,0 +1,795 @@
+  //                       TIMER CONFIGURATION                           //
+  //                    ─────────────────────────                        //
+  //                                                                     //
+  // - timer_clash_               - Uses clash_timeout from PropBase.    //
+  //                                Debounce to prevent false Clashes.   //
+  // - timer_random_event_        - Interval timer for Random Hazards.   //
+  //                                Controls how often Hazards can occur.//
+  // - timer_hazard_delay_        - Delay inbetween event trigger and    //
+  //                                Hazard dps. Gap for voice to finish. //
+  // - timer_hazard_after_revive_ - Cooldown after user revives.         //
+  //                                Blocks Hazards until timer is done.  //
+  // - timer_health_increase_     - Interval for Health recharge.        //
+  //                                Controls healing rate.               //
+  // - timer_armor_increase_      - Interval for Armor recharge.         //
+  //                                Controls Armor recharge rate.        //
+  //=====================================================================//
+
+  HEVTimerBase timer_clash_;
+  HEVTimerBase timer_random_event_;
+  HEVTimerBase timer_hazard_delay_;
+  HEVTimerBase timer_hazard_after_revive_;
+  HEVTimerBase timer_health_increase_;
+  HEVTimerBase timer_armor_increase_;
+
+  Hev() : PropBase() {
+    timer_clash_.configure(this->clash_timeout_);
+    timer_random_event_.configure(HEV_RANDOM_EVENT_INTERVAL_MS);
+    timer_hazard_delay_.configure(HEV_HAZARD_DELAY_MS);
+    timer_hazard_after_revive_.configure(HEV_HAZARD_AFTER_REVIVE_MS);
+    timer_health_increase_.configure(HEV_HEALTH_INCREASE_MS);
+    timer_armor_increase_.configure(HEV_ARMOR_INCREASE_MS);
+  }
+
+  const char* name() override { return "Hev"; }
+
+  int health_ = 100;
+  int armor_ = 100;
+
+  enum Hazard {
+    HAZARD_NONE = 0,
+    HAZARD_BIO = 1,
+    HAZARD_RAD = 2,
+    HAZARD_BLO = 3,
+    HAZARD_CHE = 4,
+    HAZARD_HEA = 5,
+    HAZARD_SHO = 6,
+  };
+
+  Hazard current_hazard_ = HAZARD_NONE;
+
+  // Calculate Physical Damage
+  void DoDamage(int damage, bool quiet = false) {
+    int tens = health_ / 10;
+    if (armor_ >= damage) {
+      armor_ -= ceilf(damage * 0.80 / 2);
+      health_ -= ceilf(damage * 0.20);
+
+    } else if (armor_ > 0) {
+      int excess_physical = damage - armor_;
+      health_ -= excess_physical;
+      // Make armor compromised sound play before setting armor to 0
+      SaberBase::DoEffect(EFFECT_USER2, 0.0);
+      PVLOG_NORMAL << "Armor Compromised!\n";
+      armor_ = 0;
+
+    } else { health_ -= damage; }
+
+    // Enforce minimum values
+    if (health_ < 0) health_ = 0;
+    if (armor_ < 0) armor_ = 0;
+
+    // Damage
+    if (!quiet) SaberBase::DoEffect(EFFECT_STUN, 0.0);
+    
+    // Death Sound
+    if (health_ == 0) {
```

Comment:
Quirk #1

We can fix this by only playing the death sound when health_ went from a non-zero value, to a zero value.
Basically, at the beginning of this function, we ave the health in a local variable, something like:

```cpp
   int previous_health = health;
```

Then here, we replace this if statement with:

```cpp
    if (health_ == 0 && prevous_health > 0) {
```

That way, if we were already out of health when DoDamage is called, the death sound will not play again.